ipn/ipnstate/ipnstate.go

```diff
@@ -183,6 +185,29 @@ type PeerStatusLite struct {
 	NodeKey key.NodePublic
 }
 
+// UserID is a tailcfg.UserID that is JSON encoded as a string.
+type UserID tailcfg.UserID
+
+// MarshalJSON marshals a UserID as a JSON string.
+func (u UserID) MarshalJSON() ([]byte, error) {
+	return fmt.Appendf(nil, `"%d"`, u), nil
+}
+
+// UnmarshalJSON unmarshals a UserID from a JSON string.
+func (u *UserID) UnmarshalJSON(b []byte) error {
+	if len(b) < 2 || b[0] != '"' || b[len(b)-1] != '"' {
```

Comment:
I thought part of the objective was to handle either numeric of string encoding, though. Otherwise isn't this a breaking change? 

Maybe:

```go
if len(b) >= 2 && b[0] == '"' && b[len(b)-1] != '"' {
  fail
} else v, err := strconv.Atoi(string(b[1:len(b)-1])); err != nil {
  fail
}
return stconv.Atoi(string(b))
```

or something like that?

Reply:
We could have it support either, but I don't think it's necessary.  I believe this is only used locally as part of LocalAPI, so it's encoded and decoded as the same string type on both sides.

Reply:
I see—in that case, we could do this more easily by implementing `text.Unmarshaler` instead. `json.Marshal` will call it with an unmarshaled string value already resolved.

See https://pkg.go.dev/encoding/json#Unmarshal amid the comments.

Reply:
TIL!  yeah, that's a lot simpler. Thanks!

Reply:
hahaha, I'm a dork.  This could be handled simply be adding a tag, there's no need for a new type:

``` go
type PeerStatus struct {
    UserID  tailcfg.UserID `json:",string"`
}
```

Reply:
Good point—I almost suggested that earlier, but then I got rabbitholed thinking we wanted to support both formats 😆